### ISSUE TITLE
Forced volume mount in Docker image

### ISSUE THREAD

**Issue by hhromic** [gliderlabs/logspout#471]
Hello,

The current Logspout Docker image contains the following Dockerfile instruction:

    VOLUME /mnt/routes

This causes Docker to **always** mount a volume for the container into the `/mnt/routes` mount-point, regardless of the user desiring or not to persist ephemeral routes.

According to the README file, one would assume that mounting a volume for persisting routes is *opt-in* instead of forced:

> Routes are stored on disk, so by default routes are ephemeral. You can mount a volume to `/mnt/routes` to persist them.

The issue with forcing a volume is that all Logspout containers start to pollute the Docker volumes subsystem with unnecessary volumes (if the user does not require persistence). For example:

```
$ docker volume ls
DRIVER              VOLUME NAME
local               artifactory_data
local               b1d471d60ec5676c173b729f1db87a7e3f540539c3986ba3be11887c924cace6
local               loki_data
local               prometheus_data
local               grafana_data
local               portainer_data
```

The more Logspout instances you have, the more of these unnamed volumes you get. Moreover, users that want to mount the volume in a different location than `/mnt/routes` and use `ROUTESPATH` are also affected by this pollution.

Please note that at this time it is not possible to "unset" a `VOLUME` instruction during container-instantiation in Docker. Therefore there is no way to avoid this unless having your own custom `Dockerfile` for Logspout without the `VOLUME` instruction.

I believe that we can safely **remove this instruction here in upstream** instead and avoid users having to create custom images that don't have functional differences except disabling the volume.

Users that actually want to mount a volume will not see any difference anyway.

Thoughts?

**Comment by hhromic** [comment]
Hi @michaelshobbs , sorry to ping you :)

Would you have some time to take a look on this issue?
It is the last remaining issue from `logspout` in our deployment that needs a bit of attention.

If you are okay with the proposed change I can make a PR for you as well. Thanks!

**Comment by michaelshobbs** [comment]
TL;DR I have questions on why but would accept a PR with specific requirements.

I've been thinking about this off and on since this issue was opened. I'd *prefer* not to remove this as it easily gives us a sane default for route persistence. Since this concern has not been brought up before, I'd like to understand better why having this default mount is a problem in your environment. So a couple questions:

- What is the problematic effect of having these unnamed volumes?
- Why run multiple logspout instances on a single docker instance? (logspout is supposed to be able to send all logs from a docker instance to whatever logging services the user would like)


Regarding removing the default mount. I would accept a PR that removes the mount and:
- maintains the default behavior
- tests the default and overridden `ROUTESPATH` behaviors
- updates the documentation with a usable example

**Comment by hhromic** [comment]
Hello, sorry for the long delay coming back to this. I forgot to reply because we are not using Logspout anymore and switched to [Vector](https://vector.dev/) to implement our use-case and also to unify other observability use-cases we work with. I will anyway answer your questions below for closure.

> * What is the problematic effect of having these unnamed volumes?

The problem is more on the management point-of-view (i.e. humans) than technical. Volume management starts to become harder due to the clutter and hard to read volume names. This generates visual noise specially on deployments with many services.

> Why run multiple logspout instances on a single docker instance? (logspout is supposed to be able to send all logs from a docker instance to whatever logging services the user would like)

Our environments have multiple nodes running in Docker Swarm mode. We used Logspout (and now Vector) to collect container logs from each node in each swarm. Then, each Logsput used to send the logs to a central Loki instance. However, Logspout still needed to be deployed distributed (`mode: global`).

In some Docker images that we use (e.g. Grafana and Loki), instead of having a forced `VOLUME` instruction in the Dockerfile, the images resort to a default _in-memory_ or _volatile_ persistence used to get you starting first. Once you are familiar with the application, you are usually instructed on how to mount the volume if you want persistence.

Thanks anyway for Logspout, it worked quite well for us in our initial approach for capturing logs from Docker!

**Comment by hhromic** [comment]
I just found this very nice article that summarises and explains the issues with specifying `VOLUME` in Dockerfiles much better than I did: https://boxboat.com/2017/01/23/volumes-and-dockerfiles-dont-mix/
